fixham/internal/docker/gobuild.go

```diff
@@ -0,0 +1,269 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultGoBuildName    = "calico/go-build"
+	defaultGoBuildVersion = "v0.91"
+
+	modCacheDir = "/go/pkg/mod"
+	goCacheDir  = "/go-cache"
+)
+
+// GoBuildImage is the struct for the go-build image
+// i.e calico/go-build:<version>
+type GoBuildImage struct {
+	imageName    string
+	imageVersion string
+}
+
+// NewGoBuild returns a new GoBuildImage
+// with default image name and version
+func NewGoBuild(version string) GoBuildImage {
+	if version == "" {
+		version = defaultGoBuildVersion
```

Comment:
It is better to return an error when `version` isn't set rather than returning a default value. calico/go-build v0.91 isn't special. It happens to be the version we current use and will get outdated pretty soon. Setting default values to a system isn't always good when they are not sane.

We frequently update go-build versions in `metadata.mk` file. Can we read from there?

Reply:
I have updated it. This tool will not use Makefiles but the value can be updated in [config](https://github.com/projectcalico/calico/pull/8935/files#diff-a8f99f44b58022c30358fd5f26c2a1ca5c1580c07c1cad437a9bb913b731497aR6)

Reply:
>  We frequently update go-build versions in metadata.mk file. Can we read from there?

Agree - we should read this from `metadata.mk`, at least for now. We should not have two places to configure this.

Longer term perhaps we can switch this to go the other way, so that this is defined in the Go code and nowhere else, but we need to make sure we only have a single place this value is configured at any given time.

Reply:
updated to read from `metadata.mk`